Enable HDF5 testing on Cray systems when cray-hdf5-parallel is loaded

daviditen opened the pull request:
Enable HDF5 testing on Cray systems when the module cray-hdf5-parallel is
loaded. Skip the directory otherwise. Use the presence of the environment
variable CRAY_HDF5_PARALLEL_VERSION as the indicator.

ronawho commented:
I'm not familiar with hdf5, so this is likely an ignorant question, but what's the difference between hdf5 and hdf5-parallel? Is this just the MPI-IO support?

Do all our tests require hdf5-parallel, or would most of them work with regular hdf5? It seems unfortunate to limit these tests to just crays. Could we instead use `h5cc -showconfig` to determine this?

e.g. for tests that will work with any version of hdf5 a skipf like:

```sh
#!/usr/bin/env bash

if h5cc -showconfig &>/dev/null; then
    echo "False"
else
    echo "True"
fi
```

And for tests that need at hdf5 parallel, can you do something like (not positive that I have the right search string for the support you need):

```sh
#!/usr/bin/env bash

if h5cc -showconfig 2>/dev/null | grep -q -i "parallel hdf5: yes"; then
    echo "False"
else
    echo "True"
fi
```

daviditen commented:
> I'm not familiar with hdf5, so this is likely an ignorant question, but what's the difference between hdf5 and hdf5-parallel? Is this just the MPI-IO support?

Yes, the difference is the inclusion of parallel operations using MPI-IO.

I updated the directory-level skipif to use your h5cc command for regular HDF5, and added your `grep parallel hdf5: yes` to the specific tests that require it.

daviditen commented:
Thanks much @ronawho!
